File: src/main/scala/vpzomtrrfrt/creepernuggets/CreeperNuggets.scala
package vpzomtrrfrt.creepernuggets

import net.minecraft.entity.item.EntityItem
import net.minecraft.entity.monster.EntityCreeper
import net.minecraft.item.ItemStack
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.event.entity.living.LivingDropsEvent
import net.minecraftforge.fml.common.Mod.EventHandler
import net.minecraftforge.fml.common.event.{FMLInitializationEvent, FMLPreInitializationEvent}
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import net.minecraftforge.fml.common.registry.GameRegistry.ItemStackHolder
import net.minecraftforge.fml.common.{Mod, SidedProxy}

import scala.util.Random

@Mod(name = CreeperNuggets.NAME, modid = CreeperNuggets.MODID, modLanguage = "scala")
object CreeperNuggets {
  final val NAME = "Creeper Nuggets"
  final val MODID = "creepernuggets"

  @SidedProxy(clientSide = "vpzomtrrfrt.creepernuggets.ClientProxy",
    serverSide = "vpzomtrrfrt.creepernuggets.CommonProxy")
  var proxy: CommonProxy = _

  @EventHandler
  def preInit(event: FMLPreInitializationEvent): Unit = {
    proxy.preInit()
    MinecraftForge.EVENT_BUS.register(this)
  }

  @EventHandler
  def init(event: FMLInitializationEvent): Unit = {
    proxy.init()
  }

  @SubscribeEvent
  def onDrops(event: LivingDropsEvent): Unit = {
    if(event.entity.isInstanceOf[EntityCreeper]) {
      event.drops.add(new EntityItem(event.entity.worldObj, event.entity.posX, event.entity.posY, event.entity.posZ, new ItemStack(ItemCreeperNugget, new Random().nextInt(4))))
    }
  }
}

File: src/main/java/vpzomtrrfrt/creepernuggets/CreeperNuggets.java
package vpzomtrrfrt.creepernuggets;

import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.loot.v1.FabricLootPoolBuilder;
import net.fabricmc.fabric.api.loot.v1.event.LootTableLoadingCallback;
import net.minecraft.entity.EntityType;
import net.minecraft.loot.UniformLootTableRange;
import net.minecraft.loot.entry.ItemEntry;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class CreeperNuggets implements ModInitializer {
    public static final String MODID = "creepernuggets";

    private static final Identifier CREEPER_LOOT_TABLE = EntityType.CREEPER.getLootTableId();

    @Override
    public void onInitialize() {
        Registry.register(Registry.ITEM, new Identifier(MODID, "creeper_nugget"), ItemCreeperNugget.INSTANCE);

        LootTableLoadingCallback.EVENT.register((resourceManager, manager, id, supplier, setter) -> {
            if(CREEPER_LOOT_TABLE.equals(id)) {
                supplier.pool(
                        FabricLootPoolBuilder.builder()
                                .rolls(UniformLootTableRange.between(0, 3))
                                .with(ItemEntry.builder(ItemCreeperNugget.INSTANCE))
                );
            }
        });
    }
}
